engines/composer/composer.cpp

```diff
@@ -93,6 +93,17 @@ Common::Error ComposerEngine::run() {
 		_directoriesToStrip = 0;
 	}
 
+	if (getPlatform() == Common::kPlatformMacintosh) {
+		const Common::FSNode gameDataDir(ConfMan.get("path"));
+		if (gameId == "sleepingcub")
+			SearchMan.addSubDirectoryMatching(gameDataDir, "sleepcub");
```

Comment:
It is better to remove these and remove them from the `directoryGlobs` and add to the individual files in the detection entry like this: https://github.com/scummvm/scummvm/blob/master/engines/director/detection_tables.h#L1904

You also need to specify `_flags = kADFlagMatchFullPaths;` like this: https://github.com/scummvm/scummvm/blob/master/engines/director/detection.cpp#L83

Reply:
Any plans to improve it?

Reply:
This hunk is not about detection but about fixing error() when running the game. Fixing it otherwise is cumbersome as we need to track where the files are through the code. I didn't touch directoryGlobs so I'd prefer it to be left out of scope of this PR